### PR TITLE
Slow down the work navbar does, and cancel updating it when receiving a high volume of user events

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
I couldn't perceive any issue changing this.  And it helps us not hit the UI thread too often while doing something like scrolling quickly.  There's little value in having the nav bars have to update that quickly, especially as it forces a UI thread switch that we're seeing in UI delays.

### CODE REVIEW

File: src/EditorFeatures/Core/NavigationBar/NavigationBarController.cs [hunk 1]
```diff
@@ -157,8 +157,6 @@ void IDisposable.Dispose()
         _eventSource.Disconnect();
 
         _disconnected = true;
-
-        // Cancel off any remaining background work
         _cancellationTokenSource.Cancel();
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/NavigationBar/NavigationBarController.cs:160]:
```suggestion

        // Cancel off any remaining background work
        _cancellationTokenSource.Cancel();
```

File: src/EditorFeatures/Core/NavigationBar/NavigationBarController.cs [hunk 2]
```diff
@@ -64,7 +64,7 @@ internal partial class NavigationBarController : IDisposable
     /// compute the model once for every batch.  The <c>bool</c> type parameter isn't used, but is provided as this
     /// type is generic.
     /// </summary>
-    private readonly AsyncBatchingWorkQueue<bool, NavigationBarModel?> _computeModelQueue;
+    private readonly AsyncBatchingWorkQueue<VoidResult, NavigationBarModel?> _computeModelQueue;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/NavigationBar/NavigationBarController.cs:67]: switched to VoidResult to make it clear this value isn't used.

File: src/EditorFeatures/Core/NavigationBar/NavigationBarController.cs [hunk 3]
```diff
@@ -93,15 +92,15 @@ internal partial class NavigationBarController : IDisposable
...
             DelayTimeSpan.Short,
             ComputeModelAndSelectItemAsync,
-            EqualityComparer<bool>.Default,
+            EqualityComparer<VoidResult>.Default,
             asyncListener,
             _cancellationTokenSource.Token);
 
         _selectItemQueue = new AsyncBatchingWorkQueue(
-            DelayTimeSpan.NearImmediate,
+            DelayTimeSpan.Short,
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/NavigationBar/NavigationBarController.cs:103]: changed the cadence on this, couldn't observe any problem at this speed.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**